### ISSUE TITLE
How-to/Can-we use serverless for modifying existing cloudformation stack?

### ISSUE THREAD

**Issue by shubsaini09** [serverless/serverless#6694]
Let's say, I have deployed my lambda and API gateway using cloud formation. Now I want to start using the serverless framework, so how do I point my new serverless template to point to existing cloud formation stack to make changes? Is it possible? If yes then how? Can someone please give a small example or reference docs? 

If this happens successfully then I am going to use to this in my production so let me know of any worst-case failure scenarios also. Will my template delete everything and recreate or it will just override on my existing stack?

Thanks for helping : )

**Comment by shubsaini09** [comment]
After hours of R&D and work,

Internally, Serverless converts to Cloud-Formation template. That means, if we are able to design a similar template, with all the resources, then we can use Serverless to modify our existing resources and they will be only updated and no recreated again. 

I tried simulating in a lower environment and below is what I observed. 

There is a way how Serverless creates the template internally and if you have not used a similar approach then you won't be able to use Serverless to modify your resources. Instead, you will end up creating new replicas of each which in turn will throw an error. Because the same name resources are not allowed in most of the places in AWS. 

Little deeper - Cloudformation does something like 'difference between old and new script' based on which it decides what to delete and what to add. This is why Serverless will create new because it won't match with existing template names. 

However, we can pick-pocket the names from our existing template and a little bit of reverse engineering can be used to do all of this. But this is all expensive in terms of time and efforts. The best way is to delete and recreate if possible. I know it's not very supportive in production environments. 

In short, the answer is no. We can't use it easily. 

I am keeping it opened for the main people to decide on this. 

Thanks

**Comment by gligorkot** [comment]
I really need this, and have been able to produce a serverless stack to exactly replicate my CF resources, but I am stuck at a problem as my current stack doesn't have a ServerlessDeploymentBucket which obviously Serverless needs for its deployment.

I'll try and work around it if I can, but haven't found anything at the moment.

**Comment by gligorkot** [comment]
Right, working through this got me to these two pages:

This one - which would be ideal - #4840 - but has been closed as something unlikely to happen... But an interesting discussion was there about being able to specify an existing deployment bucket. So I created a new bucket where serverless can upload the newly created CF Template json and this solved that issue.

I then followed the documentation here: [https://serverless.com/framework/docs/providers/aws/guide/deploying/](url), added the custom `deploymentBucket` config under `provider` and serverless successfully updated my previously manually uploaded CF Stack.

Success!

Hope this helps you or anyone else who stumbles upon this thread.
